ddtrace/encoding.py

```diff
@@ -73,8 +74,11 @@ def __init__(self):
         self.content_type = 'application/msgpack'
 
     def _encode(self, obj):
-        return msgpack.packb(obj, use_bin_type=True)
-
+        # use_bin_type kwarg only exists since msgpack-python v0.4.0
```

Comment:
that approach is correct! but this evaluation doesn't change among time execution and can be verified at import time (like we're doing for the Msgpack | JSON switch at the top of this module). Can you move that logic somewhere [here][1], so that we're not checking the `msgpack` version during every payload encoding?

Other than that it's good, and thanks for the test matrix!

[1]: https://github.com/DataDog/dd-trace-py/pull/245/files#diff-40980052c6d8dd89e59d4b1b648882fdR12